litebird_sim/mbs/mbs.py

```diff
@@ -0,0 +1,498 @@
+            self.make_fg = False
+            log.info("Foreground maps will NOT be generated")
+        if self.make_fg:
+            self.fg_models = config_mbs['fg']
+            del self.fg_models['make_fg']
+            self.fg_components = list(self.fg_models.keys())
+        try: self.out_dir = config_mbs['output']['output_directory']
+        except KeyError: self.out_dir = './'
+        log.info(f'saving files in {self.out_dir}')
+        try: self.out_string = config_mbs['output']['output_string']
+        except KeyError:
+            self.out_string = f'date_{date.today().strftime("%y%m%d")}'
+            log.info('no file_string defined, today date is set as string', rank)
+
+    def generate_noise(self):
+        instr = self.instrument
+        nmc_noise = self.nmc_noise
+        nside = self.nside
+        npix = hp.nside2npix(nside)
+        root_dir = self.out_dir
+        out_dir = f'{root_dir}/noise/'
+        seed_noise = self.seed_noise
+        N_split = self.N_split
+        file_str = self.out_string
+        channels = instr.keys()
+        N_channels = len(channels)
+        parallel = self.parallel_mc
+        col_units = [self.units, self.units, self.units]
+        rank = 0
+        size = 1
+        if parallel:
+            comm = lbs.MPI_COMM_WORLD
+            rank = comm.Get_rank()
+            size = comm.Get_size()
+        if not os.path.exists(out_dir) and rank==0 and self.save:
+                os.makedirs(out_dir)
+        nmc_noise = math.ceil(nmc_noise/size)*size
+        if nmc_noise!=self.nmc_noise:
+            log.info(f'WARNING: setting nmc_noise = {nmc_noise}', rank)
+        perrank = nmc_noise//size
+        chnl_seed = 12
+        if not self.save:
+            noise_map_matrix = np.zeros((N_channels, 3, npix))
+        for Nchnl, chnl in enumerate(channels):
+                chnl_seed += 67
+                P_sens = instr[chnl]['P_sens']
+                P_rms = from_sens_to_rms(P_sens, nside)
+                T_rms = P_rms/np.sqrt(2.)
+                tot_rms = np.array([T_rms, P_rms, P_rms]).reshape(3,1)
+                for nmc in range(rank*perrank, (rank+1)*perrank):
+                    if seed_noise:
+                        np.random.seed(seed_noise+nmc+chnl_seed)
+                    nmc_str = str(nmc).zfill(4)
+                    if (self.save) and not (os.path.exists(out_dir+nmc_str)):
+                            os.makedirs(out_dir+nmc_str)
+                    if N_split:
+                        split_rms = tot_rms*np.sqrt(N_split)
+                        noise_map = np.zeros((3, npix))
+                        for hm in range(N_split):
+                            noise_map_split = np.random.randn(3, npix)*split_rms
+                            noise_map += noise_map_split
+                            file_name = f'{chnl}_noise_SPLIT_{hm+1}of{N_split}_{nmc_str}_{file_str}.fits'
+                            file_tot_path = f'{out_dir}{nmc_str}/{file_name}'
```

Comment:
It's better to make the code OS-agnostic; the slash character `/` does not work under Windows. Use the `Path` object from the `pathlib` library:

```python
from pathlib import Path

# ...

nmc_dir = Path(f'{out_dir}{nmc_str}')

# The operator '/' is defined on paths and it concatenates two paths in
# an OS-independent way. Thus, it will be expanded to '/' under Unix
# systems, and to `\` under Windows
file_tot_path = nmc_dir / file_name
```